### ISSUE TITLE
Lost connection when subscribe

### ISSUE THREAD

**Issue by Rigogp2610** [hannesa2/paho.mqtt.android#500]
Using this library, I connect to the AWS broker and can also publish. But when I subscribe, it disconnects me. It returns the following error: java.io.EOFException.
Any ideas?
Thank you so much

**Comment by hannesa2** [comment]
I've no clue. No log, no running sample ... I can do here nothing

**Comment by Rigogp2610** [comment]
This is my code:

Gradle: 
implementation 'com.github.hannesa2:paho.mqtt.android:4.0'
implementation 'org.eclipse.paho:org.eclipse.paho.android.service:1.1.1'

Code:
fun connect() {
	try {
            val mCaCrtFile: InputStream = application.applicationContext.getAssets().open("ca.txt")
            val mCrtFile: InputStream = application.applicationContext.getAssets().open("cert.txt")
            val mKeyFile: InputStream = application.applicationContext.getAssets().open("key.txt")

            val socketFactory = getSocketFactory(mCaCrtFile, mCrtFile, mKeyFile, "")

          
            val client = MqttAndroidClient(application.applicationContext, host, clientId,
                Ack.AUTO_ACK, MemoryPersistence())

            var options = MqttConnectOptions()
            options.isAutomaticReconnect = true

            options.socketFactory = socketFactory
            options.keepAliveInterval = 15
            options.connectionTimeout = 30

            client.setCallback(object: MqttCallbackExtended {
                override fun connectionLost(cause: Throwable?) {
                }

                override fun messageArrived(topic: String?, message: MqttMessage?) {
                }

                override fun deliveryComplete(token: IMqttDeliveryToken?) {
                }

                override fun connectComplete(reconnect: Boolean, serverURI: String?) {
                    if (reconnect) subscribeToTopic(client)
                }

            })


            client.connect(options, null, object: IMqttActionListener {
                override fun onSuccess(asyncActionToken: IMqttToken?) {
                    subscribeToTopic(client)
                }

                override fun onFailure(asyncActionToken: IMqttToken?, exception: Throwable?) {
                }

            })
        } catch (e: java.lang.Exception) {
            e.printStackTrace()
        }
}

fun subscribeToTopic(client: MqttAndroidClient) {
	client.subscribe(subscriptionTopic, QoS.AtLeastOnce.value, null, object : IMqttActionListener {
		override fun onSuccess(asyncActionToken: IMqttToken) {
		}

		override fun onFailure(asyncActionToken: IMqttToken?, exception: Throwable?) {
		}
	})
}

**Comment by Rigogp2610** [comment]
This is my sample:

https://github.com/Rigogp2610/PahoMqttTest.git

Anything else you need tell me. Thanks!

**Comment by Rigogp2610** [comment]
I have already solved the problem. I have changed some parameters and I have changed to the 'org.eclipse.paho:org.eclipse.paho.client.mqttv3:1.2.0' library. With the same parameters in the other library, I get the following error: MqttException (0) - java.io.IOException: WebSocket Response header: Incorrect upgrade. I also get that error with the latest version of the paho library: 'org.eclipse.paho:org.eclipse.paho.client.mqttv3:1.2.5'
